nixos/tests/matrix/synapse-workers.nix

```diff
@@ -0,0 +1,53 @@
+import ../make-test-python.nix ({ pkgs, ... }: {
+  name = "matrix-synapse-workers";
+  meta = with pkgs.lib; {
+    maintainers = teams.matrix.members;
+  };
+
+  nodes = {
+    homeserver =
+      { pkgs
+      , nodes
+      , ...
+      }: {
+        services.postgresql = {
+          enable = true;
+          initialScript = pkgs.writeText "synapse-init.sql" ''
+            CREATE ROLE "matrix-synapse" WITH LOGIN PASSWORD 'synapse';
+            CREATE DATABASE "matrix-synapse" WITH OWNER "matrix-synapse"
+            TEMPLATE template0
+            LC_COLLATE = "C"
+            LC_CTYPE = "C";
+          '';
+        };
+
+        services.matrix-synapse = {
+          enable = true;
+          settings = {
+            database = {
+              name = "psycopg2";
+              args.password = "synapse";
+            };
+            enable_registration = true;
+            enable_registration_without_verification = true;
+
+            federation_sender_instances = [ "federation_sender" ];
+          };
+          configureRedisLocally = true;
+          workers = {
+            enable = true;
+            config = {
+              "federation_sender" = { };
+            };
+          };
+        };
+      };
+  };
+
+  testScript = ''
+    start_all()
+
+    homeserver.wait_for_unit("matrix-synapse.service");
+    homeserver.wait_for_unit("matrix-synapse-worker-federation_sender.service");
```

Comment:
Looking through the upstream documentation I'm pretty sure we can't use the health check endpoint for this as it always returns 200 OK (yes, even if the redis connection fails) and I don't see any useful metrics that could be used for this either.

Reply:
Done.

Reply:
> Done. 

I assume this was meant to be in another thread?

But yeah, seems as if there's not much to check unfortunately :/ 
I _may_ try out implementing a more detailed test, but this checks at least if everything starts up correctly and since there don't seem to be more meaningful healthchecks that are trivial to implement, let's keep it that way for now.

One thing though: can you please add this to `passthru.tests` of the synapse package? That way it gets built whenever the synapse package is touched.